src/main/java/edu/harvard/iq/dataverse/dataaccess/StorageIO.java

```diff
@@ -594,10 +594,14 @@ public static String getDriverPrefix(String driverId) {
         return driverId+ DataAccess.SEPARATOR;
     }
     
+    public static boolean isDirectUploadEnabled(String driverId) {
+        return Boolean.getBoolean(System.getProperty("dataverse.files." + driverId + ".download-redirect", "false"));
+    }
+    
     //Check that storageIdentifier is consistent with store's config
     //False will prevent direct uploads
     protected static boolean isValidIdentifier(String driverId, String storageId) {
-        return true;
+        return false;
```

Comment:
Is this going to cause a problem for SwiftAccessIO because it doesn't implement the method so now the DataAccess line 367 will return false instead of true?

Reply:
Swift doesn't do direct uploads so this nominally doesn't affect the way Swift has been used. This will prevent someone turning direct-upload on/using the /add file endpoints as a workaround for Swift until there's a check added for the Swift store. It will also prevent any new store types from allowing direct upload unless they implement a type-specific isValidIdentifier method. (Note that true was only introduced in the Remote Store PR and the intent was to flip this to false once file and S3 also had stricter validation tests in place.)